src/ardupilot/ArduPilotProtocol.cpp

```diff
@@ -0,0 +1,111 @@
+#include "ArduPilotProtocol.h"
+
+#include "../Constants.h"
+#include "../Globals.h"
+#include "../log.h"
+#include "../utils/json.h"
+
+#include <mutex>
+
+namespace net {
+namespace ardupilot {
+
+using json = nlohmann::json;
+using val_t = json::value_t;
+using namespace robot::types;
+using namespace navtypes;
+using std::placeholders::_1;
+
+ArduPilotProtocol::ArduPilotProtocol() {}
+
+void ArduPilotProtocol::initArduPilotServer() {
+	auto ardupilot_protocol = std::make_unique<net::websocket::WebSocketProtocol>(
+		Constants::ARDUPILOT_PROTOCOL_NAME);
+	ardupilot_protocol->addMessageHandler(
+		"arduPilotGPSReport", std::bind(&ArduPilotProtocol::handleGPSRequest, this, _1),
+		validateGPSRequest);
+	ardupilot_protocol->addMessageHandler(
+		"arduPilotIMUReport", std::bind(&ArduPilotProtocol::handleIMURequest, this, _1),
+		validateIMURequest);
+	ardupilot_protocol->addMessageHandler(
+		"arduPilotHeadingReport",
+		std::bind(&ArduPilotProtocol::handleHeadingRequest, this, _1), validateHeadingRequest);
+	ardupilot_protocol->addConnectionHandler(
+		std::bind(&ArduPilotProtocol::clientConnected, this));
+	ardupilot_protocol->addDisconnectionHandler(
+		std::bind(&ArduPilotProtocol::clientDisconnected, this));
+
+	Globals::websocketServer.addProtocol(std::move(ardupilot_protocol));
+}
+
+void ArduPilotProtocol::clientConnected() {
+	log(LOG_INFO, "ArduPilot connected.\n");
+	{
+		std::lock_guard<std::mutex> lock(_connectionMutex);
+		_arduPilotProtocolConnected = true;
+	}
+	_connectionCV.notify_one();
```

Comment:
Looks like we're notifying this cv but not waiting on it anywhere. Should we just remove it?

Keeping track of the connection state is a good idea though, can we add a getter `isArduPilotConnected()` or something?